### ISSUE TITLE
can not start jupyter kernel - raises NotImplementedError in asyncio\subprocess.py

### ISSUE THREAD

**Issue by vasantapte** [shroominic/codeinterpreter-api#61]
Looks like there is something wrong with below code and how jupyter kernel is started under windows 10
        self.subprocess = await asyncio.create_subprocess_exec(
            "jupyter",
            "kernelgateway",
            "--KernelGatewayApp.ip='0.0.0.0'",
            f"--KernelGatewayApp.port={self.port}",
            stdout=out,
            stderr=out,
            cwd=".codebox"
        )

>>>error log 
Starting kernel...
2023-08-01 23:06:15.682 Uncaught app exception
Traceback (most recent call last):
  File "D:\myPython\codeInterpreter\lib\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 552, in _run_script
    exec(code, module.__dict__)
  File "D:\myPython\codeInterpreter\frontend\app.py", line 37, in <module>
    asyncio.run(get_images(input_text, files=uploaded_files_list))
  File "C:\Users\Amidamaru2\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Amidamaru2\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "D:\myPython\codeInterpreter\frontend\utils.py", line 22, in get_images
    async with CodeInterpreterSession(model='gpt-3.5-turbo') as session:
  File "D:\myPython\codeInterpreter\codeinterpreterapi\session.py", line 204, in __aenter__
    await self.astart()
  File "D:\myPython\codeInterpreter\codeinterpreterapi\session.py", line 39, in astart
    await self.codebox.astart()
  File "D:\myPython\codeInterpreter\lib\site-packages\codeboxapi\box\localbox.py", line 119, in astart
    self.subprocess = await asyncio.create_subprocess_exec(
  File "C:\Users\Amidamaru2\AppData\Local\Programs\Python\Python310\lib\asyncio\subprocess.py", line 218, in create_subprocess_exec
    transport, protocol = await loop.subprocess_exec(
  File "C:\Users\Amidamaru2\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1681, in subprocess_exec
    transport = await self._make_subprocess_transport(
  File "C:\Users\Amidamaru2\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 498, in _make_subprocess_transport
    raise NotImplementedError
NotImplementedError

**Comment by forestmoster** [comment]
mee  too window 10,all version is newest

**Comment by shroominic** [comment]
Ahh I think is has something todo with the differnce between Unix/Windows terminal.
Maybe someone with windows can fix this and start a PR to codeboxapi

**Comment by yutaro38** [comment]
I am facing the same bug. This bug seems to remain in v0.0.11.

**Comment by tjroamer** [comment]
> Ahh I think is has something todo with the differnce between Unix/Windows terminal. Maybe someone with windows can fix this and start a PR to codeboxapi

After some tries, in app.py, I replaced

```python
asyncio.run(get_images(input_text, files=uploaded_files_list))
```

by

```python
    loop = asyncio.ProactorEventLoop()
    asyncio.set_event_loop(loop)
    loop.run_until_complete(get_images(input_text, files=uploaded_files_list))
```

The reported error was gone, but I faced another issue: 

```
[W 2023-08-23 07:51:44.881 KernelGatewayApp] Permission to listen on port 8888 denied
[I 2023-08-23 07:51:44.883 KernelGatewayApp] Jupyter Kernel Gateway 2.5.2 is available at http://0.0.0.0:8889
Waiting for kernel to start...
Waiting for kernel to start...
Waiting for kernel to start...
Waiting for kernel to start...
Waiting for kernel to start...
Waiting for kernel to start...
Waiting for kernel to start...
Waiting for kernel to start...

```
After a deeper look at the error message, I found the cause is that in `codeboxapi/box/localbox.py` an exception `aiohttp.ServerDisconnectError` was raised, I have no idea how to solve it though.

```python
        while True:
            try:
                response = await self.aiohttp_session.get(self.kernel_url)
                if response.status == 200:
                    break
            except aiohttp.ClientConnectorError:
                pass
            except aiohttp.ServerDisconnectedError:
                pass
            if settings.VERBOSE:
                print("Waiting for kernel to start...")
            await asyncio.sleep(1)
```
